research-software-directory/RSD-as-a-service
# 1055 - show reference papers and scraped citations

# Show reference papers and scraped citations

Closes #1055

Changes proposed in this pull request:
* Reference papers section is added to software page (image 1)
* Scraped citations of reference papers are integrated into mentions section of software page
* Mention count RPC is updated to include scraped citations (total mention count)
* Mention count in software card is updated (using updated mention count RPC)
* Unrelated change: upgraded postgrest to latest version to fix 416 response error when using `count=exact` 

How to test:

* `make start` to build solution and generate test data
*  login as rsd-user and create test software. Or login as rsd-admin and modify existing one.
*  Navigate to reference papers section and add following reference papers: 
   *  doi:10.1186/s13321-017-0220-4
   * 10.1186/1758-2946-6-3
* Navigate to mentions and add few mentions via import and manually
* Confirm reference papers are show
* Confirm all mentions are shown. Note! in order to let scraper run in the background you will need to restart if you are used make start. Use docker compose down and then docker compose up (observe logs for scraper service). 
* Confirm mention counts are correct on software page and in the card of software overview page 
* Try adding some of scraped mentions manually in the mentions section. This is allowed but only one entry will be shown and counted

## Reference papers on software page
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/3a7fc6b4-09f3-447d-9ee2-cd5797d2b1ca)
## Reference papers on edit software page
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/ab219379-c1be-461a-9e78-794507b76ee2)

## Mentions on edit software page
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/3567aed2-e1bb-4158-bc4d-10e5dbfb317c)

## Mentions count on software page (consolidated count of mentions and citations)
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/e72b63d4-0f1b-4afe-a76c-41cd4fa37366)

## Mentions count in the software card (and order on mention count)
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/9d8dccc9-d0ac-4d0b-a4bc-b01bc0a64871)


PR Checklist:
